lib/motion/revision_calculator.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "digest"
+require "motion"
+
+module Motion
+  class RevisionCalculator
+    attr_reader :revision_paths
+
+    def initialize(revision_paths:)
+      @revision_paths = revision_paths
+    end
+
+    def perform
+      derive_file_hash
+    end
+
+    private
+
+    def derive_file_hash
+      digest = Digest::MD5.new
+
+      files.each do |file|
+        digest << File.read(file)
+      end
+
+      digest.hexdigest
+    end
+
+    def existent_paths
+      @existent_paths ||=
+        begin
+          revision_paths.all_paths.flat_map(&:existent)
+        rescue
+          raise BadRevisionPathsError
+        end
+    end
+
+    def existent_files(path)
+      Dir["#{path}/**/*", path].reject { |f| File.directory?(f) }.uniq
```

Comment:
:+1: Smart idea using `directory?` instead of `file?`. This will correctly handle symlinks.